cni-plugin/pkg/dataplane/linux/dataplane_linux.go

```diff
@@ -128,14 +136,16 @@ func (d *linuxDataplane) DoNetworking(
 			if err != nil {
 				return err
 			}
-			err = disableDAD(hostVethName)
+			err = hostNS.Do(func(_ ns.NetNS) error {
+				return disableDAD(hostVethName)
+			})
```

Comment:
I wonder if it would work to postpone this step and the following `LinkSetUp` until exit from this `ns.WithNetNSPath` context?  I guess it depends on whether the container side of the veth can get IPv6 addressing while the host side is still down.  If the postponement worked, I think that would mean:
- not needing the host-side disableDAD here, and instead probably integrate that sysctl into configureSysctls
- not needing the host-side disable_ipv6 addition just below, as it would be covered by the existing configureSysctls logic.

Reply:
Also looks like we currently (i.e. pre-existing) have a small window where the host-side veth could accept an IPv6 RA (router advertisement).  It the host-side LinkSetUp could be postponed, it would eliminate that window.

Reply:
Not sure if it is possible, I think we need the host link local address when setting up the routes for container veth: https://github.com/projectcalico/calico/blob/03b78052790cb845d186bd1103e7fe15ce55c665/cni-plugin/pkg/dataplane/linux/dataplane_linux.go#L256

We might be able to move configureSysctls for hostVeth before LinkSetUp is called.

Reply:
I moved the configureSysctls call before the link is up.

Reply:
I modified the code to get rid of host-side disableDAD and merge it in configureSysctls.